### ISSUE TITLE
 linkerd 2.6.1: intermittent 502 bad gateway 

### ISSUE THREAD

**Issue by dilshad18** [linkerd/linkerd2#4630]
### What is the issue?

Cross namespace communication is blocked by linked and curl shows 502 bad gateway.
 linked shows 0% success rate in dashboard 

## linkerd-controller's Logs ##
2020-05-15T04:52:46.526841698Z time="2020-05-15T04:52:46Z" level=warning msg="Found stats for unknown route: xxx.svc.cluster.local:"
2020-05-15T04:52:46.52684642Z time="2020-05-15T04:52:46Z" level=warning msg="Found stats for unknown route: xxx.svc.cluster.local:"
2020-05-15T04:52:46.526850122Z time="2020-05-15T04:52:46Z" level=warning msg="Found stats for unknown route: xxx.svc.cluster.local:"
2020-05-15T04:52:46.526854519Z time="2020-05-15T04:52:46Z" level=warning msg="Found stats for unknown route: xxx.svc.cluster.local:"
2020-05-15T04:52:48.517617455Z time="2020-05-15T04:52:48Z" level=warning msg="Found stats for unknown route: xxx.svc.cluster.local:"
2020-05-15T04:52:48.517636009Z time="2020-05-15T04:52:48Z" level=warning msg="Found stats for unknown route: xxx.svc.cluster.local:"
2020-05-15T04:52:48.517656911Z time="2020-05-15T04:52:48Z" level=warning msg="Found stats for unknown route: xxx.svc.cluster.local:"
2020-05-15T04:52:48.517659893Z time="2020-05-15T04:52:48Z" level=warning msg="Found stats for unknown route: xxx.svc.cluster.local:"
2020-05-15T04:52:48.517662323Z time="2020-05-15T04:52:48Z" level=warning msg="Found stats for unknown route: xxx.svc.cluster.local:"
2020-05-15T04:52:50.507595162Z time="2020-05-15T04:52:50Z" level=warning msg="Found stats for unknown route: xxx.svc.cluster.local:"

#### `linkerd check` output

```text
⇒  linkerd check
kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

kubernetes-version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

linkerd-existence
-----------------
√ 'linkerd-config' config map exists
√ heartbeat ServiceAccount exist
√ control plane replica sets are ready
√ no unschedulable pods
√ controller pod is running
√ can initialize the client
√ can query the control plane API

linkerd-config
--------------
√ control plane Namespace exists
√ control plane ClusterRoles exist
√ control plane ClusterRoleBindings exist
√ control plane ServiceAccounts exist
√ control plane CustomResourceDefinitions exist
√ control plane MutatingWebhookConfigurations exist
√ control plane ValidatingWebhookConfigurations exist
√ control plane PodSecurityPolicies exist

linkerd-identity
----------------
√ certificate config is valid
√ trust roots are using supported crypto algorithm
√ trust roots are within their validity period
√ trust roots are valid for at least 60 days
√ issuer cert is using supported crypto algorithm
√ issuer cert is within its validity period
√ issuer cert is valid for at least 60 days
√ issuer cert is issued by the trust root

linkerd-api
-----------
√ control plane pods are ready
√ control plane self-check
√ [kubernetes] control plane can talk to Kubernetes
√ [prometheus] control plane can talk to Prometheus
√ tap api service is running

linkerd-version
---------------
√ can determine the latest version
‼ cli is up-to-date
    is running version 2.7.1 but the latest stable version is 2.8.0
    see https://linkerd.io/checks/#l5d-version-cli for hints

control-plane-version
---------------------
‼ control plane is up-to-date
    is running version 2.6.1 but the latest stable version is 2.8.0
    see https://linkerd.io/checks/#l5d-version-control for hints
‼ control plane and cli versions match
    control plane running stable-2.6.1 but cli running stable-2.7.1
    see https://linkerd.io/checks/#l5d-version-control for hints

Status check results are √
```

### Environment

- Kubernetes Version: v1.14.9
- Cluster Environment: AWS EKS
- Host OS: AWS (GNU/Linux x86_64 )
- Linkerd version: 2.6.1

### Possible solution
Deletion of linkerd-destination pod fix the problem

**Comment by olix0r** [comment]
@dilshad18 A number of known bugs have been fixed since stable-2.6.1. Are you able to upgrade to `stable-2.8.0`? If not, I suggest at least upgrading the proxy on the affected workloads via the annotation `config.linkerd.io/proxy-version: stable-2.8.0`

**Comment by dilshad18** [comment]
will upgrade linkerd.Thanks for your suggestion.
